packages/sanity/src/core/preview/utils/getPreviewStateObservable.ts

```diff
@@ -94,13 +109,25 @@ export function getPreviewStateObservable(
   )
 
   return combineLatest([draft$, published$, version$, versions$]).pipe(
-    map(([draft, published, version, versions]) => ({
-      draft: draft.snapshot ? {title, ...(draft.snapshot || {})} : null,
-      isLoading: false,
-      published: published.snapshot ? {title, ...(published.snapshot || {})} : null,
-      version: version.snapshot ? {title, ...(version.snapshot || {})} : null,
-      versions,
-    })),
+    map(([draft, published, version, versions]) => {
+      let v = version.snapshot ? {title, ...(version.snapshot || {})} : null
+
+      /**
+       * when the search type should be raw, there are versions and the document provided is a version
+       * then we need to find the exact version to display
+       */
+      if (perspective.isRaw && versions && isVersionId(documentId)) {
+        v = {title, ...versions[getVersionFromId(documentId) ?? '']?.snapshot}
+      }
```

Comment:
Could this be part of the `version$` observable?

Reply:
Maybe! I'll try to have a look and see if I can put it in there :)

Reply:
Here we go, done here [refactor(sanity): update to array prop vs string](https://github.com/sanity-io/sanity/pull/7815/commits/fe34b34a6299959a7bd0fd6e9f9c7f9ea76e361d) :) 

Thanks for the suggestion!